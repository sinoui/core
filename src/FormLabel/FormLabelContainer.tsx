import React from 'react';
import styled from 'styled-components';
import Label from '@sinoui/core/Label';

/**
 * 表单项中的标签容器组件属性
 *
 * @export
 * @interface FormLabelContainerProps
 */
export interface FormLabelContainerProps {
  /**
   * 标签文本。
   */
  children: React.ReactNode;
  /**
   * 标签宽度
   */
  width?: number | string;
  /**
   * label要绑定元素的id
   */
  name?: string;
  /**
   * 文本对齐方式
   */
  align?: 'left' | 'right' | 'center' | 'justify';
  /**
   * 文本之后添加冒号
   */
  colon?: boolean;
  /**
   * 警告状态
   */
  warning?: boolean;
  /**
   * 错误状态
   */
  error?: boolean;
  /**
   * 焦点状态
   */
  focused?: boolean;
  /**
   * 不可用状态
   */
  disabled?: boolean;
  /**
   * 必填项
   */
  required?: boolean;
  /**
   * label要绑定元素的id
   */
  htmlFor?: string;
  /**
   * 是否垂直显示
   *
   * @type {boolean}
   * @memberof FormLabelContainerProps
   */
  vertical?: boolean;
}

/**
 * @private
 * 表单标签容器
 */
const FormLabelContainer = styled(Label).attrs(
  ({ htmlFor, name }: FormLabelContainerProps) => ({
    htmlFor: htmlFor || name,
  }),
)<FormLabelContainerProps>`
  display: flex;
  justify-content: space-between;
  width: ${(props) => (props.vertical ? '100%' : props.width || '120px')};
  ${(props) => !props.vertical && 'min-height: 36px;'};
  box-sizing: border-box;
  padding: 4px 8px;
  padding-left: 0;
  ${(props) =>
    props.colon &&
    `
    &::after {
      content: ":";
      margin: 0 8px 0 2px;
    }
  `}
`;

export default FormLabelContainer;
