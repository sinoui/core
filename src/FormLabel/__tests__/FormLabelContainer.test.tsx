import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import '@testing-library/jest-dom/extend-expect';
import FormLabelContainer from '../FormLabelContainer';

it('正确渲染FormLabelContainer', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <FormLabelContainer>标签容器</FormLabelContainer>
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('垂直布局的标签容器', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <FormLabelContainer vertical>标签容器</FormLabelContainer>
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('带冒号的标签', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <FormLabelContainer align="right" colon>
          标签容器
        </FormLabelContainer>
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
