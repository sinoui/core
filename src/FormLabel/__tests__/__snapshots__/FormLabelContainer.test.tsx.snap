// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`垂直布局的标签容器 1`] = `
<label
  className="Label-sc-4jto8h-0 FormLabelContainer-iuornh-0 ckPTLN"
>
  标签容器
</label>
`;

exports[`带冒号的标签 1`] = `
<label
  className="Label-sc-4jto8h-0 FormLabelContainer-iuornh-0 cCbUWe"
>
  标签容器
</label>
`;

exports[`正确渲染FormLabelContainer 1`] = `
<label
  className="Label-sc-4jto8h-0 FormLabelContainer-iuornh-0 gJOjYe"
>
  标签容器
</label>
`;
